spec.md

```diff
@@ -0,0 +1,461 @@
+# The UltraStar File Format
+
+The UltraStar file format is a timed text format for describing karaoke songs where singing performance is scored by the karaoke software.
+
+## Status of this document
+
+This document aims to describe the UltraStar File Format in its most recent version 2.0.0.
+
+> [!IMPORTANT]
+>
+> This document is currently a work-in-progress. Parts of the final specification may change significantly from the current state of this document.
+
+GitHub Issues are preferred for discussion of this specification. Alternatively, you can discuss comments on our Discord server.
+
+## 1. Introduction
+
+The UltraStar file format provides a standardized representation of karaoke songs. The format has been used for a long time by numerous karaoke games such as UltraStar Deluxe, Performous, or Vocaluxe. There exists an ecosystem of supporting applications for hosting, editing, and managing songs. However due to the lack of an official file format specification implementations differ and new features cannot be added in a consistent manner. This document aims to fix this problem by providing a formal specification for the syntax and semantics of the UltraStar file format.
+
+The UltraStar file format is designed to be edited by machines and humans alike and is intended to be easily understood and edited by technical and non-technical users. This guiding principle is influential for many decisions made during the design process.
+
+### 1.1. Conventions in this Document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+The grammatical rules in this document are to be interpreted as described in [RFC5234](https://datatracker.ietf.org/doc/html/rfc5234). We are using the following core rules:
+
+```abnf
+CR    = %x0D     ; carriage return
+LF    = %x0A     ; line feed
+CRLF  = CR LF
+DIGIT = %x30-39  ; 0-9
+HTAB  = %x09     ; horizontal tab
+SP    = %x20     ; space
+```
+
+
+
+## 2. General Structure
+
+UltraStar files are plain text files. The UTF-8 encoding MUST be used. Implementations MUST NOT add a byte order mark to the beginning of a file. In the interests of interoperability, implementations MAY ignore the presence of a byte order mark rather than treating it as an error.
+
+> [!CAUTION]
+>
+> Whether files with a BOM may be accepted is an open discussion (#46).
```

Comment:
I thought GitHub might auto-link the issues, but apparently it didn't. I added explicit links to the discussion issues. It might be a good idea to open issues for the other undecided questions as well so that we have a place to discuss these things. Do you think that would be a good idea that I open those ~12 issues now and link them in the spec?